components/camera/rtsp/rtsp.go

```diff
@@ -162,21 +164,55 @@ func (rc *rtspCamera) reconnectClient() (err error) {
 			}
 		}
 	}()
-	err = rc.client.Start(rc.u.Scheme, rc.u.Host)
+	// send Start to server URL and timeout if you can't
+	doneStart := make(chan struct{})
+	go func() {
+		err = rc.client.Start(rc.u.Scheme, rc.u.Host)
```

Comment:
does this client not have a ctx? not sure why this needs to be in a goroutine. we don't want this to block forever

Reply:
test was failing because the client was sending a DESCRIBE request to an RTSP server and for some reason never got a response. So I put it in a gorountine such that the following select{} statement could time out if it the server never responded. But you're right that the goroutine should exit if the response never comes too, and needs a context.